polkadot/xcm/pallet-xcm/src/lib.rs

```diff
@@ -2432,6 +2438,74 @@ impl<T: Config> Pallet<T> {
 		AccountIdConversion::<T::AccountId>::into_account_truncating(&ID)
 	}
 
+	pub fn dry_run_call<Runtime, Router, OriginCaller, RuntimeCall>(
```

Comment:
Missing docs :P At least it could mention that this should be used by the runtime api or similar.